flink-core/src/test/java/org/apache/flink/api/common/typeutils/TypeSerializerUpgradeTestBase.java

```diff
@@ -421,7 +421,7 @@ public void upgradedSerializerIsValidWhenCompatibleAsIs() throws Exception {
         DataInputView serializedData =
                 readAndThenWriteData(dataInput, serializer, serializer, testDataMatcher);
         TypeSerializerSnapshot<T> snapshot = writeAndThenReadSerializerSnapshot(serializer);
-        TypeSerializer<T> restoreSerializer = snapshot.restoreSerializer();
+        TypeSerializer<T> restoreSerializer = snapshot.restoreSerializer().duplicate();
```

Comment:
I think it would be better to check both cases only `restoreSerializer()` and with `duplicate()`.